code/modules/heavy_vehicle/equipment/cult.dm

```diff
@@ -0,0 +1,33 @@
+/obj/item/mecha_equipment/mounted_system/soul_javelin
+	name = "mounted soul javelin"
+	desc = "A heavy duty daemon shardlauncher, not for the faint of heart."
+	icon_state = "mecha_souljavelin"
+	holding_type = /obj/item/gun/energy/rifle/cult/mounted
+	equipment_delay = -8
+	restricted_hardpoints = list(HARDPOINT_LEFT_HAND, HARDPOINT_RIGHT_HAND)
+	restricted_software = list(MECH_SOFTWARE_CULT)
+
+/obj/item/mecha_equipment/doomblade
+	name = "daemon doomblade"
+	desc = "A large blade menacing with demonic energy, try to not touch the red parts."
+	icon_state = "mecha_doomblade"
+	restricted_hardpoints = list(HARDPOINT_LEFT_HAND, HARDPOINT_RIGHT_HAND)
+	restricted_software = list(MECH_SOFTWARE_CULT)
+	origin_tech = list(TECH_MATERIAL = 2, TECH_ENGINEERING = 2)
+	var/obj/item/melee/cultblade/doomblade
+
+/obj/item/mecha_equipment/doomblade/Initialize()
+	. = ..()
+	doomblade = new /obj/item/melee/cultblade/mounted(src)
+
+/obj/item/mecha_equipment/doomblade/attack(mob/living/M, mob/living/user)
+	if(!owner)
+		return
+	var/target_zone
+	for(var/mob/living/carbon/human/pilot in owner.pilots)
```

Comment:
This will interact weirdly if we have multiple pilots in the mech ever. Might pick the wrong pilot, not the one that is using the blade.

Why isn't it just something like...

```suggestion
	target_zone = BP_CHEST
	if(user in owner.pilots)
		if(user.zone_sel.selecting)
			target_zone = user.zone_sel.selecting
```

Reply:
i believe "user" in this case, is the mech, so that wouldn't work.

Reply:
Hmmm. Maybe this isn't possible then with current mechcode.

Reply:
I know you merged this already, but mechs do have a target zone that is set when you click.